interfaces/cython/cantera/onedim.pyx

```diff
@@ -837,6 +928,121 @@ cdef class Sim1D:
+
+            # create extra columns
+            for e in extra:
+               if e == 'grid':
+                   extra_cols[e] = self.grid
+               else:
+                   extra_cols[e] = self.profile(dom, e)
+            if self.radiation_enabled:
+                extra_cols['qdot'] = self.radiative_heat_loss
+
+            states = T, P, np.vstack(Y).T
+            return states, extra_cols, meta
+
+        elif isinstance(dom, Inlet1D):
+            self.gas.TPY = dom.T, self.P, dom.Y
+
+            # create extra columns
+            for e in extra:
+               if e == 'velocity':
+                   extra_cols[e] = dom.mdot/self.gas.density
+               elif e not in {'lambda'}:
+                   extra_cols[e] = getattr(dom, e)
+
+            return self.gas.TPY, extra_cols, meta
+
+        elif isinstance(dom, ReactingSurface1D):
+            return dom.surface.TPY, extra_cols, meta
+
+        elif isinstance(dom, Boundary1D):
+            return ([], [], []), extra_cols, meta
+
+        else:
+            msg = ("Export of '{}' is not implemented")
+            raise NotImplementedError(msg.format(type(self).__name__))
+
+    def restore_data(self, domain, states, extra_cols, meta):
+        """
+        Restore data vector of domain *domain* from `SolutionArray` *arr*.
+
+        Derived classes set default values for *domain* and *extra*, where
+        defaults describe flow domain and essential non-thermodynamic solution
+        components of the configuration, respectively. An alternative *domain*
+        (e.g. inlet, outlet, etc.), can be specified either by name or the
+        corresponding Domain1D object itself.
+        """
+        idom = self.domain_index(domain)
+        dom = self.domains[idom]
+        T, P, Y = states
+
+        if isinstance(dom, _FlowBase):
+            grid = extra_cols['grid']
+            dom.grid = grid
+            xi = (grid - grid[0]) / (grid[-1] - grid[0])
+            self._get_initial_solution()
+
+            # restore temperature and 'extra' profiles
+            self.set_profile('T', xi, T)
+            for key, val in extra_cols.items():
+                if key in ['grid', 'qdot']:
+                    pass
+                elif key in dom.component_names:
+                    self.set_profile(key, xi, val)
+
+            # restore species profiles
+            for i, spc in enumerate(self.gas.species_names):
+                self.set_profile(spc, xi, Y[:, i])
+
+            # restore pressure
+            self.P = P[0]
+
+            # restore settings
+            dom.settings = meta
+
+        elif isinstance(dom, Inlet1D):
+            self.gas.TPY = T, P, Y
+            dom.T = T
+            self.P = P
+            dom.Y = Y
+            dom.mdot = extra_cols['velocity'] * self.gas.density
+
+        elif isinstance(dom, ReactingSurface1D):
+            dom.surface.TPY = T, P, Y
+
+        elif isinstance(dom, Boundary1D):
+            pass
```

Comment:
It might be better to be explicit here about the `Boundary1D` classes that do work without any additional state. Otherwise, this would inadvertently let through any new `Boundary1D` derived classes that should require some state variables to be set.